src/main/java/org/spongepowered/api/text/serializer/TextTemplateConfigSerializer.java

```diff
@@ -0,0 +1,131 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.serializer;
+
+import com.google.common.reflect.TypeToken;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.objectmapping.ObjectMappingException;
+import ninja.leaping.configurate.objectmapping.serialize.TypeSerializer;
+import org.spongepowered.api.text.LiteralText;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.TextTemplate;
+import org.spongepowered.api.text.format.TextFormat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a {@link TypeSerializer} for {@link TextTemplate}s. TextTemplates
+ * are serialized in two parts.
+ *
+ * <p>First, the template's arguments as defined by
+ * {@link TextTemplate#getArguments()} are serialized to the "arguments" node.
+ * This is where the argument definitions are kept.</p>
+ *
+ * <p>Second, the template's text representation as defined by
+ * {@link TextTemplate#toText()} is serialized to the "content" node.</p>
+ *
+ * <p>Deserialization is a bit more complicated. We start by loading the
+ * "content" Text and check the root Text element as well as it's children. If
+ * a {@link LiteralText} value is found that is wrapped in curly braces we
+ * check to see if the value inside the braces is defined as an argument in the
+ * "arguments" nodes. If so, we use the name and format from the original
+ * LiteralText and obtain whether the argument is optional from the definition.
+ * This is repeated until there are no more Text children to check and we
+ * return a TextTemplate of the elements we have collected.</p>
+ */
+public class TextTemplateConfigSerializer implements TypeSerializer<TextTemplate> {
+
+    private static final String NODE_CONTENT = "content";
+    private static final String NODE_ARGS = "arguments";
+    private static final String NODE_OPT = "optional";
+    private static final String NODE_OPEN_ARG = "openArg";
+    private static final String NODE_CLOSE_ARG = "closeArg";
+
+    @Override
+    public TextTemplate deserialize(TypeToken<?> type, ConfigurationNode value) throws ObjectMappingException {
+        String openArg = value.getNode(NODE_OPEN_ARG).getString(TextTemplate.DEFAULT_OPEN_ARG);
+        String closeArg = value.getNode(NODE_CLOSE_ARG).getString(TextTemplate.DEFAULT_CLOSE_ARG);
+        Text content = value.getNode(NODE_CONTENT).getValue(TypeToken.of(Text.class));
+        List<Object> elements = new ArrayList<>();
+        elements.add(content.getFormat());
```

Comment:
Adding note for myself: Need to recurse through *all* children for each Text object including children's children, children's children's children, etc.